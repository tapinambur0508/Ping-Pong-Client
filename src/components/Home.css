#home .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#home .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#home .wrapper nav {
  margin: 10px 0;
}

#home .wrapper nav a.btn {
  border-radius: 10px;
}