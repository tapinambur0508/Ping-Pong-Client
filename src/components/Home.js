import React from 'react';
import NavLink from 'react-router-dom/NavLink';

import './Home.css';

export default () => (
  <section id="home">
    <div className="container">
      <div className="wrapper">
        <h1 className="heading">Ping-Pong</h1>
        <nav>
          <NavLink to="/start-game" className="btn btn-success btn-block">
            <i className="fas fa-play mr-2"></i>Start Game
            </NavLink>
          <NavLink to="/profile" className="btn btn-info btn-block">
            <i class="far fa-user-circle mr-2"></i>Profile
          </NavLink>
        </nav>
      </div>
    </div>
  </section>
);
