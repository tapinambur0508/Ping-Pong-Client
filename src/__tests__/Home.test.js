import React from 'react';
import Home from '../components/Home';
import NavLink from 'react-router-dom/NavLink';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

describe('Home Component', () => {
  it('should be defined', () => {
    expect(Home).toBeDefined();
  });

  it('Home Component matches the snapshot', () => {
    const context = {
      props: {}
    };
    const tree = renderer.create(
      <StaticRouter location="home" context={context}>
        <Home />
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  it('Count NavLink should be equal two', () => {
    const wrapper = shallow(<Home />);
    const links = wrapper.find(NavLink);

    expect(links).toHaveLength(2);
  });
});
