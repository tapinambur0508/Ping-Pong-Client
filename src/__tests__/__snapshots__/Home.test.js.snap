// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Home Component matches the snapshot 1`] = `
<section
  id="home"
>
  <div
    className="container"
  >
    <div
      className="wrapper"
    >
      <h1
        className="heading"
      >
        Ping-Pong
      </h1>
      <nav>
        <a
          aria-current={null}
          className="btn btn-success btn-block"
          href="/start-game"
          onClick={[Function]}
        >
          <i
            className="fas fa-play mr-2"
          />
          Start Game
        </a>
        <a
          aria-current={null}
          className="btn btn-info btn-block"
          href="/profile"
          onClick={[Function]}
        >
          <i
            className="far fa-user-circle mr-2"
          />
          Profile
        </a>
      </nav>
    </div>
  </div>
</section>
`;
